src/app/(Authentication)/register/page.tsx

```diff
@@ -0,0 +1,212 @@
+import { registerUser } from '@/redux/slices/RegisterSlice';
+import { RootState } from '@/redux/store';
+import { useAppDispatch, useAppSelector } from '@/redux/hooks/hook';
+
+export interface FormDataInterface {
+  name: string;
+  email: string;
+  password: string;
+  phone: string;
+  address: string;
+}
+
+export interface ErrorInterface {
+  target: RegistrationKeys;
+  msg: string;
+}
+
+export type RegistrationKeys = keyof FormDataInterface;
+
+const InitialFormValues: FormDataInterface = {
+  name: '',
+  email: '',
+  password: '',
+  phone: '',
+  address: ''
+};
+
+export interface RegisterError {
+  status: 'Success!' | 'Error';
+  message: string;
+  data: { field: string; message: string }[];
+}
+
+export default function Register() {
+  const router = useRouter();
+  const dispatch = useAppDispatch();
+  const { loading, success, error } = useSelector(
+    (state: RootState) => state.register
+  );
+
+  const [formData, setFormData] =
+    useState<FormDataInterface>(InitialFormValues);
+  const [errors, setErrors] = useState<ErrorInterface[]>([]);
+  const [openAlert, setOpenAlert] = useState<boolean>(false);
+
+  const handleChange = (key: RegistrationKeys, value: string) => {
+    setErrors([]);
+    setFormData(prevFormData => {
+      const updatedFormData = { ...prevFormData, [key]: value };
+      validateField(key, value);
+      return updatedFormData;
+    });
+  };
+
+  const validateField = (key: RegistrationKeys, value: string) => {
+    const newErrors = [...errors.filter(error => error.target !== key)];
+    if (key === 'password') {
+      const error = testPassword(value) || '';
+      newErrors.push({ target: key, msg: error });
+    } else if (!regExPatterns[key].test(value)) {
+      const errorMsg =
+        RegistrationFields.find(field => field.key === key)?.message ||
+        'Invalid input';
+      newErrors.push({ target: key, msg: errorMsg });
+    }
+    setErrors(newErrors);
+  };
+
+  const getErrorForField = (key: RegistrationKeys) => {
+    const error = errors.find(error => error.target === key);
+    return error ? error.msg : null;
+  };
+
+  const handleSubmit = async (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    setErrors([]);
+    const newErrors: ErrorInterface[] = [];
+
+    for (const key in formData) {
+      const typedKey = key as RegistrationKeys;
+      if (!regExPatterns[typedKey].test(formData[typedKey])) {
+        const errorMsg =
+          RegistrationFields.find(field => field.key === typedKey)?.message ||
+          'Invalid input';
+        newErrors.push({ target: typedKey, msg: errorMsg });
+      }
+    }
+
+    if (newErrors.length > 0) {
+      setErrors(newErrors);
+      return;
+    }
+
+    const result = await dispatch(registerUser(formData));
+
+    if (registerUser.fulfilled.match(result)) {
+      swal({
+        title: 'Registered Successfully!',
+        text: 'Please verify your email!',
+        icon: 'success',
+        timer: 2000
```

Comment:
why do we have those two alert ?? this one and the last one on this page

Reply:
I have removed the alert on the bottom 👍